classes/Kohana/Cache/Memcache.php

```diff
@@ -301,6 +298,10 @@ public function _failed_request($hostname, $port)
 			// We're looking at the failed server
 			if ($hostname == $server['host'] and $port == $server['port'])
 			{
+				// Make sure that we do not disable servers that have `instant_death` set to `FALSE`.
+				if ( ! $server['instant_death'])
+					continue;
```

Comment:
Thanks @sigginet. I would have preferred an early `return` here, but there won't be thousands of Memcache servers configured I guess, so it's good.